src/main/resources/META-INF/resources/primefaces/photocam/photocam.js

```diff
@@ -1163,6 +1202,33 @@ PrimeFaces.widget.PhotoCam = PrimeFaces.widget.BaseWidget.extend({
         } else {
             PrimeFaces.error('Capture error: AdvancedPhotoCam not attached to the camera');
         }
+    },
+    
+    /**
+     *  Retrieves the available video input device list 
+     *  @return {Promise<InputDeviceInfo[]>} Returns a promise to resolve the enumeration or null if the browser doesn't support media devices enumeration
+     */
+    getAvailableDevices: function() {
+        var result = null;
+        if (navigator.mediaDevices && navigator.mediaDevices.enumerateDevices) {
+            result = navigator.mediaDevices.enumerateDevices().then(devices => devices.filter(
```

Comment:
Lambdas are nice and easier to write, but unfortunately not supported by all target browsers yet, and not used anywhere else in the codebase afaik. Can you change that to `function(devices) { ... }`

Reply:
@blutorange is correct. Only ECMASCRIPT5 features are used and supported in PF.